org_fedora_oscap/rule_handling.py

```diff
@@ -937,49 +950,55 @@ def eval_rules(self, ksdata, storage, report_only=False):
                                         common.MESSAGE_TYPE_INFO, msg))
 
         # services, that should be added
-        services_to_add = (svc for svc in self._add_svcs
-                           if svc not in ksdata.firewall.services)
+        new_services_to_add = (svc for svc in self._add_svcs
+                               if svc not in firewall_proxy.EnabledServices)
 
         # ports, that should be added
-        ports_to_add = (ports for ports in self._add_ports
-                        if ports not in ksdata.firewall.ports)
+        new_ports_to_add = (ports for ports in self._add_ports
+                            if ports not in firewall_proxy.EnabledPorts)
 
         # trusts, that should be added
-        trusts_to_add = (trust for trust in self._add_trusts
-                         if trust not in ksdata.firewall.trusts)
+        new_trusts_to_add = (trust for trust in self._add_trusts
+                             if trust not in firewall_proxy.Trusts)
 
-        for svc in services_to_add:
+        for svc in new_services_to_add:
             # add the service unless already added
             if not report_only:
                 self._added_svcs.add(svc)
-                ksdata.firewall.services.append(svc)
 
             msg = _("service '%s' has been added to the list of services to be "
                     "added to the firewall" % svc)
             messages.append(RuleMessage(self.__class__,
                                         common.MESSAGE_TYPE_INFO, msg))
+        if not report_only:
+            all_services = list(self._add_svcs.union(set(firewall_proxy.EnabledServices)))
```

Comment:
Would it be better to put together add `self._added_svcs` to `firewall_proxy.EnabledServices`? We know they will be disjoint sets.

Reply:
I would say that it's the same.